homeassistant/components/duotecno/climate.py

```diff
@@ -0,0 +1,79 @@
+"""Support for Duotecno climate devices."""
+from typing import Any
+
+from duotecno.unit import SensUnit
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, HVACMODE, PRESETMODES
+from .entity import DuotecnoEntity, api_call
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Duotecno climate based on config_entry."""
+    cntrl = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        DuotecnoClimate(channel) for channel in cntrl.get_units(["SensUnit"])
+    )
+
+
+class DuotecnoClimate(DuotecnoEntity, ClimateEntity):
+    """Representation of a Duotecno climate entity."""
+
+    _unit: SensUnit
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_modes = [HVACMode.OFF, HVACMode.HEAT, HVACMode.COOL]
+    _attr_preset_modes = list(PRESETMODES)
+
+    @property
+    def current_temperature(self) -> int | None:
+        """Get the current temperature."""
+        return self._unit.get_cur_temp()
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Get the target temperature."""
+        return self._unit.get_target_temp()
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Get the current hvac_mode."""
+        return HVACMODE[self._unit.get_state()]
+
+    @property
+    def preset_mode(self) -> str:
+        """Get the preset mode."""
+        return next(
+            (key for key, val in PRESETMODES.items() if val == self._unit.get_preset()),
+            "Sun",
+        )
```

Comment:
Other integrations create a second map from all the options, this enables a quick lookup, for example Tado:
https://github.com/home-assistant/core/blob/2a443648fc5da6dbdcce5713c21ad63b70ad989d/homeassistant/components/tado/const.py#L148

This way we don't have to loop over the list everytime, afaik this would cost less and is more efficient. (and yes the example I showed you doesn't align with my last comment, imo Tado should change)